internal/dot/graph.go

```diff
@@ -60,12 +71,27 @@ type Group struct {
 	Type    reflect.Type
 	Name    string
 	Results []*Result
+	State   State
+
+	// ResultMap maps constructor ID to the grouped result it produces so we
+	// can get the result that failed given constructor ID.
+	ResultMap map[uintptr]*Result
```

Comment:
This is a bit confusing. Are these all constructors that produce values to
this value group? This is a map from those constructors to which results?
(Constructors have multiple results.)

Also, can we avoid having the 'index' here  and just have it in the `Graph`?
This can have a list of IDs of all constructors that provide to this group if
needed.

That lets us keep the book-keeping in one place.